src/y_map.rs

```diff
@@ -262,57 +261,58 @@ impl YMap {
         ValueView::new(self)
     }
 
-    pub fn observe(&mut self, f: PyObject) -> PyResult<ShallowSubscription> {
+    pub fn observe(&mut self, f: PyObject) -> PyResult<PyOrigin> {
         match &mut self.0 {
             SharedType::Integrated(v) => {
                 let doc = v.doc.clone();
-                let sub_id: SubscriptionId = v
-                    .inner
-                    .observe(move |txn: &TransactionMut, e| {
+                let origin = Origin::from(uuid_v4().to_string());
```

Comment:
Here and in similar changes around `observe | observe_deep` - creating origin from randomly generated uuid seemed to be the easiest way to generate unique origin instance.